crypto/dsa/dsa_ossl.c

```diff
@@ -176,31 +175,28 @@ static int dsa_sign_setup(DSA *dsa, BN_CTX *ctx_in,
 
     /* Compute r = (g^k mod p) mod q */
 
-    if (!BN_copy(kq, k))
-        goto err;
-
     /*
      * We do not want timing information to leak the length of k, so we
      * compute g^k using an equivalent exponent of fixed length. (This
      * is a kludge that we need because the BN_mod_exp_mont() does not
      * let us specify the desired timing behaviour.)
      */
 
-    if (!BN_add(kq, kq, dsa->q))
+    if (!BN_add(k, k, dsa->q))
         goto err;
-    if (BN_num_bits(kq) <= BN_num_bits(dsa->q)) {
-        if (!BN_add(kq, kq, dsa->q))
+    if (BN_num_bits(k) <= BN_num_bits(dsa->q)) {
+        if (!BN_add(k, k, dsa->q))
             goto err;
     }
 
-    BN_set_flags(kq, BN_FLG_CONSTTIME);
+    BN_set_flags(k, BN_FLG_CONSTTIME);
```

Comment:
I think we should move this up to just after the `while(BN_is_zero(k))` above - to minimise the time where `k` does not have the flag set to avoid future mishaps if the code is changed.